### ISSUE TITLE
tach report: ValueError: Encountered invalid filepath: ""

### ISSUE THREAD

**Issue by h0uter** [gauge-sh/tach#274]
Every folder I run `tach report` for I get this error.
```py
❯ tach report .
Traceback (most recent call last):
  File "/Users/wjmeijer/Repositories/spice/.venv/bin/tach", line 8, in <module>
    sys.exit(start())
             ^^^^^^^
  File "/Users/wjmeijer/Repositories/spice/.venv/lib/python3.12/site-packages/tach/start.py", line 10, in start
    main()
  File "/Users/wjmeijer/Repositories/spice/.venv/lib/python3.12/site-packages/tach/cli.py", line 941, in main
    tach_report(
  File "/Users/wjmeijer/Repositories/spice/.venv/lib/python3.12/site-packages/tach/cli.py", line 739, in tach_report
    report(
  File "/Users/wjmeijer/Repositories/spice/.venv/lib/python3.12/site-packages/tach/report.py", line 51, in report
    return create_dependency_report(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Encountered invalid filepath: ""
```

Workspace structure
```
.

├── .env
├── .gitignore
├── .tach
│   ├── .gitignore
│   ├── .latest-version
│   └── tach.info
├── .venv
│   └── ...
├── README.md
├── agent_app
│   ├── ...
│   └── compose.yml
├── central_app
│   ├── ...
│   ├── compose.yml
├── spice.code-workspace
├── src_repos
│   ├── project-a
│   │   ├── README.md
│   │   ├── pyproject.tool
│   │   └── src
│   │   │   └── project_a
│   ├── project-b
│   │   ├── README.md
│   │   ├── pyproject.toml
│   │   └── src
│   │   │   └── project_b
│   └── project-c
│   │   ├── README.md
│   │   ├── pyproject.toml
│   │   └── src
│   │   │   └── project_c
└── tach.toml
```

**Comment by emdoyle** [comment]
Thanks for raising this! I can reproduce the issue with `tach report .`  specifically. Does this also happen when you run `tach report src_repos/project-a/src/project-a/...` (or any other filepath)?

I suspect it is isolated to `.` as a filepath, I'll investigate and push a fix.

**Comment by h0uter** [comment]
I'd expected the "source root directory" would be where the `pyproject.toml` is located, not the `src` subfolder.

Probably related: note that I am using the ["src layout" style](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/) of setting up my project.  So that means that in my `pyproject.toml`'s there is this line:

```toml
[tool.setuptools]
package-dir = { "" = "src" }
```
This line ensures we don't have to write `from src.project_c import ...` but that we can simply write `from project_c import ...`.

In my experience, it would have been usefull to see a note [in the docs for this](https://docs.gauge.sh/usage/configuration#example-monorepo) on whether "src layout" is also supported and what the recommended setup is for that.

In summary:
I had all "source root directory" set to the folders containing the `pyproject.toml`... Perhaps I expected tach would see the `package-dir = { "" = "src" }` line in `pyproject.toml` and figure out the modules are actually located in the `src` folders. But then `tach sync` also didn't work...

Now I took your advice and set all the "source root directory" to `src` in every project and `tach sync` appears to work correctly works. So for me it was an onboarding challenge related to using ["src layout"](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/).
